src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,201 @@
+        /// <param name="channel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity of the PWM signal.</param>
+        public UnixPwmChannel(
+            int chip,
+            int channel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _chip = chip;
+            _channel = channel;
+            Validate();
+            Open();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
+
+            SetFrequency(frequency);
+            DutyCyclePercentage = dutyCyclePercentage;
+
+            // NOTE: There appears to be a lower-level bug that won't allow the
+            // polarity to be changed until a valid period has been written.
+            SetPolarity(polarity);
+        }
+
+        /// <summary>
+        /// The frequency in hertz.
+        /// </summary>
+        public override double Frequency =>_frequency;
+
+        /// <summary>
+        /// The duty cycle percentage represented as a value between 0.0 and 1.0.
+        /// </summary>
+        public override double DutyCyclePercentage
+        {
+            get
+            {
+                return _dutyCyclePercentage;
+            }
+            set
+            {
+                SetDutyCyclePercentage(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the frequency period in nanoseconds.
+        /// </summary>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <returns>The frequency period in nanoseconds.</returns>
+        private static int GetPeriodInNanoSeconds(double frequency)
+        {
+            // In Linux, the period needs to be a whole number and can't have a decimal point.
+            return (int)((1.0 / frequency) * 1_000_000_000);
+        }
+
+        // Helpers for file paths.
+        private string GetChipPath() => $"/sys/class/pwm/pwmchip{_chip}";
+        private string GetChannelPath() => $"{GetChipPath()}/pwm{_channel}";
+        private string GetEnablePath() => $"{GetChannelPath()}/enable";
+
+        /// <summary>
+        /// Sets the polarity for the channel to start in.
+        /// </summary>
+        /// <param name="polarity">The polarity to set.</param>
+        private void SetPolarity(PwmPolarity polarity)
+        {
+            string polarityString = polarity switch
+            {
+                PwmPolarity.ActiveHigh => "normal",
+                PwmPolarity.ActiveLow => "inversed",
+                _ => throw new ArgumentException($"Invalid PWM polarity {polarity}.")
+            };
+
+            File.WriteAllText($"{GetChannelPath()}/polarity", Convert.ToString(polarityString));
```

Comment:
Why do you have to call Convert.ToString if polarityString is already a string?

Reply:
oops.  copy/paste from other types.  will update